File: src/main/webapp/js/reserved.js
function check(){
    go= true;
    dateC= document.getElementById("dateC").value;
    timeC= document.getElementById("timeC").value;
    numberC= document.getElementById("numberC").value;
    // --
    dateG= document.getElementById("dateG").value;
    timeG= document.getElementById("timeG").value;
    numberG= document.getElementById("numberG").value;
    // --
    dateL= document.getElementById("dateL").value;
    timeL= document.getElementById("timeL").value;
    numberL= document.getElementById("numberL").value;

    if(!dateC.length==0||!dateG.length==0||!dateL.length==0){
        go= true;
    } else{
        alert("用餐日期未選取!");
        go= false;
        return false;
    }
    if(!timeC.length==0||!timeG.length==0||!timeL.length==0){
        go= true;
    } else{
        alert("用餐時間未選取!");
        go= false;
        return false;
    }
    if(!numberC.length==0||!numberG.length==0||!numberL.length==0){
        go= true;
    } else{
        alert("用餐人數未填寫!");
        go= false;
        return false;
    }

    if(go==true){
        alert("預約成功!");
        document.reservedForm.submit()
    }
}

File: src/main/java/Reserved.java


import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import project.DataBase;
import project.ReservedData;

/**
 * 
 * @projectName project
 * @className Reserved
 * @author allen
 * @date 2022年8月5日 下午2:58:27
 * Servlet implementation class Login
 */
@WebServlet("/Reserved")
public class Reserved extends HttpServlet {
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		request.setCharacterEncoding("utf-8");
		response.setContentType("text/html;charset=utf-8");
		
		String name= request.getParameter("username");
		String mail= request.getParameter("mail");
		String phone= request.getParameter("phone");
		String [] dateA= {"dateC", "dateG", "dateL"};
		String [] timeA= {"timeC", "timeG", "timeL"};
		String [] numberA= {"numberC", "numberG", "numberL"};
		String deptno="", date="", time="", number="";
		int i;
		
		for(i=0; i<dateA.length; i++) {
			date= request.getParameter(dateA[i]);
			time=  request.getParameter(timeA[i]);
			number= request.getParameter(numberA[i]);
			if(date.length()!=0) {
				break;
			} else {
				continue;
			}
		}
		switch(i) {
		case 0:
			deptno= "D01";
			break;
		case 1:
			deptno= "D02";
			break;
		case 2:
			deptno= "D03";
			break;
		}
		var rdata= new ReservedData(name, deptno, date, time, number, mail, phone);
		var dataBase= new DataBase();
		dataBase.insert(rdata);
		response.sendRedirect("Menu.html");
	}

}
